### DISCUSSION TITLE
Can multiple dynamic modals exist on the same page at the same time?

### DISCUSSION THREAD

**Discussion by ShookLyngs** [mantinedev/mantine#4388] [Q&A]
While writting modals with `@mantine/modals`, say I have this:
```tsx
import { openModal } from '@mantine/modals';

export function SubmitModal(props) { ... }

export function openSubmitModal(props) {
  openModal({
    ...,
    children: (<SubmitModal ... />)
  });
}

export function AlertModal(props) { ... }

export function openAlertModal(props) {
  openModal({
    ...,
    children: (<AlertModal ... />)
  });
}
```

The logic here:
1. Call `openSubmitModal` in many pages, to fill in input or other form fields
2. Inside SubmitModal, when an error occurs, call `openAlertModal` to show error message
3. When the AlertModal is closed, the state of SubmitModal is reset because when the AlertModal was opened, the SubmitModal was cleared by the modals manager

So, my questions are:
- Can multiple dynamic modals exist on app at the same time, so their state don't get cleared at all?
- If they cannot exist at the samte time, do I have to maintain local states for each dynamic modal?

**Comment by julamb** [comment]
I've stumbled on the same problem and would be interested in an answer as well. 

I could not find a way to open a new modal without dismounting the previous one, and could only think of two alternatives:
- Keep the modal local state outside of the modal children components (in a global state)
- Use the Modal component instead of the @mantine/modals manager

It would be awesome to make this configurable (i.e. have a parameter to make modals stack on top of each other instead of replacing each other), but I'm guessing it might be a big refactor.

Edit: Some previous answers by @rtivital, looks like there's no secret alternative: https://github.com/mantinedev/mantine/discussions/2527, https://github.com/mantinedev/mantine/discussions/3063
